pkg/provisioner/ironic/ironic.go

```diff
@@ -1162,21 +1162,21 @@ func (p *ironicProvisioner) GetFirmwareComponents() ([]metal3api.FirmwareCompone
 	if !p.availableFeatures.HasFirmwareUpdates() {
 		return nil, fmt.Errorf("current ironic version does not support firmware updates")
 	}
+
+	// Setting to 2 since we only support bmc and bios
+	componentsInfo := make([]metal3api.FirmwareComponentStatus, 0, 2)
+
+	bmcAccess, _ := p.bmcAccess()
```

Comment:
You cannot safely ignore this error. Especially when you build your provisioner with `provisioner.BuildHostDataNoBMC()` so that this will always fail and then crash on line 1171.

Reply:
I've removed since I created the generic error.